File: app/src/main/java/com/example/radzik/recipes/document/DocumentReader.java
package com.example.radzik.recipes.document;

/**
 * Created by Radzik on 09.09.2017.
 */

public class DocumentReader {

    private static DocumentReader mInstance = null;

    public static synchronized DocumentReader getInstance() {
        if(mInstance == null) {
            mInstance = new DocumentReader();
        }

        return mInstance;
    }

    private DocumentReader() {

    }


}

File: app/src/main/java/com/example/radzik/recipes/document/DocumentReader.kt
package com.example.radzik.recipes.document

/**
 * Created by Radzik on 09.09.2017.
 */

object DocumentReader {

    private var mInstance: DocumentReader? = null

    val instance: DocumentReader
        @Synchronized get() {
            if (mInstance == null) {
                mInstance = DocumentReader()
            }

            return mInstance
        }


}
